src/target/riscv/riscv.c

```diff
@@ -852,24 +859,26 @@ static int maybe_add_trigger_t2_t6(struct target *target,
 		struct trigger_request_info lt_1 = {
 			.tdata1 = fields.common | fields.size.any | fields.chain.enable |
 				fields.match.lt,
-			.tdata2 = trigger->address,
+			.tdata2 = trigger->address + trigger->length,
 			.tdata1_ignore_mask = fields.tdata1_ignore_mask
 		};
 		struct trigger_request_info ge_2 = {
 			.tdata1 = fields.common | fields.size.any | fields.chain.disable |
 				fields.match.ge,
-			.tdata2 = trigger->address + trigger->length,
+			.tdata2 = trigger->address,
 			.tdata1_ignore_mask = fields.tdata1_ignore_mask
 		};
 		ret = try_setup_chained_match_triggers(target, trigger, lt_1, ge_2);
 		if (ret != ERROR_TARGET_RESOURCE_NOT_AVAILABLE)
 			return ret;
 	}
 	LOG_TARGET_DEBUG(target, "trying to setup equality match trigger");
+	if (trigger->length > 1)
+		LOG_TARGET_WARNING(target, "trigger will only match accesses at address 0x%"
+				TARGET_PRIxADDR, trigger->address);
```

Comment:
Is that true? I think the spec says that a size any address trigger at 0x1001 will (should?) match on a 32-bit access at 0x1000.

select=0 says:
> There is at least one compare value and it contains the lowest virtual address of the access. In addition, it is recommended that there are additional compare values for the other accessed virtual addresses match. (E.g. on a 32-bit read from 0x4000, the lowest address is 0x4000 and the other addresses are 0x4001, 0x4002, and 0x4003.)

Reply:
Thanks for the catch!
I didn't consider this difference in 1.0 and 0.13.2 versions of the spec.

I would like to ask, is there a way for a debugger to test which of those additional compare values are supported for a particular access?
As far as I can see, the only way to do so is to write a program to some scratch memory which tries to trigger a watchpoint.
However, this is quite complex and not available on some targets.

Reply:
Addressed (Changed the message).